relay-server/src/utils/multipart.rs

```diff
@@ -185,12 +185,15 @@ where
             let mut item = Item::new(ItemType::Attachment);
             item.set_attachment_type(infer_type(field.name()));
             item.set_filename(file_name);
-            let content_type = match field.content_type() {
-                Some(string) => string.into(),
-                None => ContentType::OctetStream,
-            };
             // Extract the body after the immutable borrow on `file_name` is gone.
-            item.set_payload(content_type, field_data(&mut field, item_limit).await?);
+            if let Some(content_type) = field.content_type() {
+                item.set_payload(
+                    content_type.into(),
+                    field_data(&mut field, item_limit).await?,
+                );
+            } else {
+                item.set_truncated_payload(field_data(&mut field, item_limit).await?);
```

Comment:
Nit: While the interface parameters make more sense, the name `set_payload()` vs `set_truncated_payload()` still seems weirdly asymmetric since both truncate... but I have no better name, except for using the verbose option of `set_payload_without_content_type()` (which was the name before I created the PR, and chose the intentionally tacky variant with the `None` option). I am open to suggestions; maybe verbosity is okay if only one call site exists.

Reply:
I would prefer `set_payload_without_content_type`, for what it's worth. `set_truncated_payload()` seems to imply that `set_payload` does not truncate, as you noted.